### PR TITLE
Update gems and mysql

### PR DESCRIPTION
**Pull request by aepyornis**
This updates lots'o'gems.

Changes:
*  We have moved to using mariadb 10.1 and this pull requests updates the mysql2 gem to version 4
* ruby version upgraded to 2.3.6
* amazon sdk upgrade to version 3
* Rails bumped to latest version of the 4.2 series (an upgrade to 5 is coming soon!)
* Remove some older, unused gems

Tests on travis are now run using mariadb 10.1

It does some other housekeeping such as updating our readme and robocop.yml

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,32 +1,73 @@
...
+
+Metrics/MethodLength:
+  Exclude:
+    - 'spec/**/*.rb'
+
+Metrics/AbcSize:
+  Exclude:
+    - 'spec/**/*.rb'
+
+Metrics/CyclomaticComplexity:
```
Comment (aguestuser) [inline .rubocop.yml]: whoa! i'd love to know what cyclomatic complexity is!

File: .travis.yml [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
+  mariadb: '10.1'
+  
 services:
   - mysql
+
+before_install:
+  - sudo apt-get install software-properties-common
+  - sudo apt-key adv --recv-keys --keyserver hkp://keyserver.ubuntu.com:80 0xcbcb082a1bb943db
+  - sudo add-apt-repository 'deb [arch=amd64,i386,ppc64el] http://sfo1.mirrors.digitalocean.com/mariadb/repo/10.1/ubuntu trusty main'
+  - sudo apt-get install libmariadbclient-dev
```
Comment (aguestuser) [inline .travis.yml]: cool. :)

File: .travis.yml [hunk 3]
```diff
@@ -1,13 +1,25 @@
 language: ruby
 cache: bundler
 rvm:
-  - 2.3.3
+  - 2.3.6
+
+addons:
+  mariadb: '10.1'
```
Comment (aguestuser) [inline .travis.yml]: well that's a bump! :)

File: Gemfile [hunk 4]
```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 
-gem 'rails', '4.2.8'
-gem 'mysql2', '~> 0.3.21'
+gem 'rails', '4.2.10'
+gem 'mysql2', '~> 0.4.10'
```
Comment (aguestuser) [inline Gemfile]: bump!

File: README.md [hunk 5]
```diff
@@ -22,7 +17,7 @@ LittleSis uses these technologies among others:
 - D3
 
 
-The best way to get up and running with LittleSis development is to use docker: [LittleSis-Docker](https://github.com/aepyornis/littlesis-docker)
+The best way to get up and running with LittleSis development is to use docker: [LittleSis-Main](https://github.com/public-accountability/littlesis-main)
```
Comment (aguestuser) [inline README.md]: the rein of `littlesis-main` begins!

**Comment by lizstarin** [comment]
👍

File: .rubocop.yml (hunk 1 above)
Comment (aepyornis) [reply .rubocop.yml]: "the number of linearly independent paths through a method"

Comment (aepyornis) [reply .rubocop.yml]: detail: https://rubocop.readthedocs.io/en/latest/cops_metrics/#metricscyclomaticcomplexity

**Comment by aepyornis** [comment]
@lizstarin @aguestuser 👍 here we go!

**Pull request merged**